### ISSUE TITLE
[V3] Implement new Concept Exercise: regular-expressions

### ISSUE THREAD

**Issue by biancapower** [exercism/ruby#1061]
This issue describes how to implement the `regular-expressions` concept exercise for the Ruby track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Regular Expressions in Ruby.

## Learning objectives

- Know how to use regular expressions in Ruby

## Out of scope

- advanced regex

## Concepts

- `regular-expressions` as they are handled in Ruby

## Prerequisites

- `basics`
- `variables`

### Hints

- [Ruby docs on Regexp](https://ruby-doc.org/core-2.7.1/Regexp.html)
- [Rubular - a Ruby regular expression editor](https://rubular.com/)

### After

- [Mastering Ruby Regular Expressions](https://www.rubyguides.com/2015/06/ruby-regex/)

## Representer

This exercise does not require any specific representation logic to be added to the [representer](https://github.com/exercism/ruby-representer).

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer](https://github.com/exercism/ruby-analyzer).

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/ruby/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
